python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -0,0 +1,191 @@
+    @return a 2-d numpy.array containing the correction kernel
+
+    @note As currently implemented, kappa is a static (single) kernel.
+    """
+    def post_conv_kernel_ft2(kernel, sig1=1., sig2=1.):
+        kft = fft2(kernel)
+        return np.sqrt((sig1**2 + sig2**2) / (sig1**2 + sig2**2 * kft**2))
+
+    def post_conv_kernel2(kernel, sig1=1., sig2=1.):
+        kft = post_conv_kernel_ft2(kernel, sig1, sig2)
+        out = ifft2(kft)
+        return out
+
+    pck = post_conv_kernel2(kappa, sig1=sig2, sig2=sig1)
+    pck = ifftshift(pck.real)
+
+    # I think we may need to "reverse" the PSF, as in the ZOGY (and Kaiser) papers...
+    # This is the same as taking the complex conjugate in Fourier space before FFT-ing back to real space.
+    if False:  # TBD: figure this out. For now, we are turning it off.
+        pck = pck[::-1, :]
+
+    return pck
+
+
+def computeCorrectedDiffimPsf(kappa, im2_psf, sig1=0.2, sig2=0.2):
+    """! Compute the (decorrelated) difference image's new PSF:
+    post_conv_psf = phi_1(k) * sym.sqrt((sig1**2 + sig2**2) / (sig1**2 + sig2**2 * kappa_ft(k)**2))
+
+    @param kappa  A matching kernel array derived from Alard & Lupton PSF matching
+    @param im2_psf The uncorrected psf array of the science image (and also of the diffim)
+    @param sig1   Average sqrt(variance) of template image used for PSF matching
+    @param sig2   Average sqrt(variance) of science image used for PSF matching
+    @return a 2-d numpy.array containing the new PSF
+    """
+    def post_conv_psf_ft2(psf, kernel, sig1=1., sig2=1.):
+        # Pad psf or kernel symmetrically to make them the same size!
+        if psf.shape[0] < kernel.shape[0]:
+            while psf.shape[0] < kernel.shape[0]:
+                psf = np.pad(psf, (1, 1), mode='constant')
+        elif psf.shape[0] > kernel.shape[0]:
+            while psf.shape[0] > kernel.shape[0]:
+                kernel = np.pad(kernel, (1, 1), mode='constant')
+        psf_ft = fft2(psf)
+        kft = fft2(kernel)
+        out = psf_ft * np.sqrt((sig1**2 + sig2**2) / (sig1**2 + sig2**2 * kft**2))
+        return out
+
+    def post_conv_psf(psf, kernel, sig1=1., sig2=1.):
+        kft = post_conv_psf_ft2(psf, kernel, sig1, sig2)
+        out = ifft2(kft)
+        return out
+
+    im2_psf_small = im2_psf
+    # First compute the science image's (im2's) psf, subset on -16:15 coords
+    if im2_psf.shape[0] > 50:
+        x0im, y0im = getImageGrid(im2_psf)
+        x = np.arange(-16, 16, 1)
+        y = x.copy()
+        x0, y0 = np.meshgrid(x, y)
+        im2_psf_small = im2_psf[(x0im.max()+x.min()+1):(x0im.max()-x.min()+1),
+                                (y0im.max()+y.min()+1):(y0im.max()-y.min()+1)]
+    pcf = post_conv_psf(psf=im2_psf_small, kernel=kappa, sig1=sig2, sig2=sig1)
+    pcf = pcf.real / pcf.real.sum()
+    return pcf
+
+
+def performExposureDecorrelation(templateExposure, exposure, subtractedExposure, psfMatchingKernel, log):
```

Comment:
Please consider renaming to `decorrelateExposure`. It would be nice if you could find a name short enough that you were not tempted to rename it when using it in other code (e.g. the unit test), as that renaming can be confusing.